### PR TITLE
[4.0] AfterDelete events to return void

### PR DESCRIPTION
**Pull request by SharkyKZ**
### Summary of Changes

The return values of after delete events are not used. Unless they should be, this changes them to be void.

### Testing Instructions

Code review or try deleting some stuff. Should work like before.

### Documentation Changes Required

No.

### CODE REVIEW

File: plugins/finder/newsfeeds/newsfeeds.php [hunk 1]
```diff
@@ -116,7 +116,7 @@ public function onFinderCategoryChangeState($extension, $pks, $value)
 	 * @since   2.5
 	 * @throws  Exception on database error.
 	 */
-	public function onFinderAfterDelete($context, $table)
+	public function onFinderAfterDelete($context, $table): void
```
Comment (Quy) [inline plugins/finder/newsfeeds/newsfeeds.php:119]: Update `@return`

File: plugins/finder/tags/tags.php [hunk 2]
```diff
@@ -93,7 +93,7 @@ class PlgFinderTags extends Adapter
 	 * @since   3.1
 	 * @throws  Exception on database error.
 	 */
-	public function onFinderAfterDelete($context, $table)
+	public function onFinderAfterDelete($context, $table): void
```
Comment (Quy) [inline plugins/finder/tags/tags.php:96]: Update `@return `

**Comment by HLeithner** [comment]
Changing this behavior is a b/c break and have to be deprecated or at least documented.

Also removing a provided true and false in case of an error seams to be wrong.

**Comment by SharkyKZ** [comment]
It is already documented as void. E.g. https://docs.joomla.org/Special:MyLanguage/Plugin/Events/Content#onContentAfterDelete

>Also removing a provided true and false in case of an error seams to be wrong.

We don't use these values in core, i.e. returning `false` does nothing, unlike with `BeforeDelete` events. And 3PDs can't rely on these values as well because we don't return them consistently. If you see a use for boolean values here, we could switch to booleans instead. But then we still could not rely on them until 3PDs consistently start returning booleans as well.

**Comment by HLeithner** [comment]
Ok didn't checked the docu so seams to make sense to remove it.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on b1db89adb1f127248ba4b3825e260005f72a192e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29148">issues.joomla.org/tracker/joomla-cms/29148</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**